drake/solvers/Optimization.h

```diff
@@ -406,6 +406,10 @@ namespace Drake {
       return conlist;
     }
```

Comment:
note that all of the get..() methods immediately above are only here so that the snopt user function can access them (because it wasn't easy for me to declare that method a friend without the snopt-specific types coming into the header file).  again, i'm sure there is a better way.